src/dune_pkg/rev_store.mli

```diff
@@ -21,15 +21,33 @@ module At_rev : sig
 end
 
 module Remote : sig
+  (** [uninit] represents an uninitialized remote. Use [update] or
+      [don't_update] to get a remote [t] *)
+  type uninit
```

Comment:
I wonder why do we need this type?

Reply:
It looks like it's so people don't forget to update the remote (returned by `add_repo`) or explicitly declare that they won't update it (`don't_update`).

Reply:
Exactly this. I didn't like that it it was easily possible to accidentally forget to update and then be surprised why the `Rev_store` would never have any content.

Originally started with a phantom type as described in the classic Jane Street post about access control, but given there is no need to ever switch back and forth between the unupdated and the updated state I simplified it, to just have `uninit` and force the API consumer to use `update` or `don't_update` deliberately. It could use a `update_again : t -> t/unit Fiber.t` but given we don't ever do that this can be added if we need it.

Reply:
As I see it, `Rev_store` is basically just a database. A database isn't limited to being updated or not updated after it is initialized. It can be updated whenever and those operations can be interleaved with other operations. It doesn't seem like a limitation for us at the moment, but it does seem like something gratuitous. In any case, it's not blocking the PR.